# Time and Memory Analysis
Using [line_profiler](https://github.com/rkern/line_profiler) to analyze time, and [memory_profiler](https://github.com/fabianp/memory_profiler) to analyze memory.

In [7]:
# only going to be going through one runthrough (only for the second set of images)
!C:\Users\kyhan\Anaconda3\envs\kaggledsb\Lib\site-packages\kernprof.py -l -v time_memory_run_file.py

the program has finished
Wrote profile results to final_ram_limited.py.lprof
Timer unit: 3.8012e-07 s

Total time: 0.108285 s
File: final_ram_limited.py
Function: make_mask at line 13

Line #      Hits         Time  Per Hit   % Time  Line Contents
    13                                           @profile
    14                                           # helper functions
    15                                           def make_mask(center,diam,z,width,height,spacing,origin):
    16                                               '''
    17                                               Center : centers of circles px -- list of coordinates x,y,z
    18                                               diam : diameters of circles px -- diameter
    19                                               widthXheight : pixel dim of image
    20                                               spacing = mm/px conversion rate np array x,y,z
    21                                               origin = x,y,


  0%|          | 0/89 [00:00<?, ?it/s]


Seems to be running relatively fast, thus there probaby will not be any optimization needed.

In [8]:
!python -m memory_profiler time_memory_run_file.py

the program has finished
Filename: final_ram_limited.py

Line #    Mem usage    Increment   Line Contents
    13  211.602 MiB    0.000 MiB   @profile
    14                             # helper functions
    15                             def make_mask(center,diam,z,width,height,spacing,origin):
    16                                 '''
    17                                 Center : centers of circles px -- list of coordinates x,y,z
    18                                 diam : diameters of circles px -- diameter
    19                                 widthXheight : pixel dim of image
    20                                 spacing = mm/px conversion rate np array x,y,z
    21                                 origin = x,y,z mm np.array
    22                                 z = z position of slice in world coordinates mm
    23                                 '''
    24  211.605 MiB    0.004 MiB       mask = np.zeros([height,width]) # 0's everywhere except nodule swapping x,y to match 


  0%|          | 0/89 [00:00<?, ?it/s]


Does not seem to use massive amounts of memory, and the variables will be overwritten so it should never exceed 212 MiB. No memory edits to the code are needed.